src/server/app/stac_api_collection_search.py

```diff
@@ -0,0 +1,174 @@
+        # this makes it possible to iterate through all collections
+        self.catalog.add_conforms_to("COLLECTIONS")
+        results: List[CollectionMetadata] = []
+        all_collections = self.catalog.get_collections()
+        while len(results) < self.limit:
+            try:
+                collection = next(all_collections)
+            except StopIteration:
+                break
+
+            # check bbox overlap
+            # TODO: get min/max values from ALL bbox values
+            bbox_overlap = (
+                check_bbox_overlap(
+                    self.bbox,
+                    get_full_bounding_box(
+                        collection.extent.spatial.bboxes  # type: ignore
+                    ),
+                )
+                if self.bbox
+                else True
+            )
+
+            # check temporal overlap
+            collection_temporal_extent: Tuple[
+                Optional[datetime], Optional[datetime]
+            ] = (
+                ensure_utc(collection.extent.temporal.intervals[0][0]),
+                ensure_utc(collection.extent.temporal.intervals[0][1]),
+            )
+
+            temporal_overlap = (
+                check_datetime_overlap(
+                    self.datetime,
+                    collection_temporal_extent,
+                )
+                if self.datetime
+                else True
+            )
+
+            # check text fields for overlap
+            text_fields = [collection.id]
+            if collection.keywords:
+                text_fields.extend(collection.keywords)
+            if collection.title:
+                text_fields.append(collection.title)
+            if collection.description:
+                text_fields.append(collection.description)
+
+            text_overlap = (
+                check_text_overlap(self.text, text_fields) if self.text else True
+            )
+
+            if bbox_overlap and temporal_overlap and text_overlap:
+                hint = None
+                if self.hint_lang == PYTHON:
+                    hint = generate_pystac_client_hint(
+                        base_url=self.base_url,
+                        collection_id=collection.id,
+                        bbox=self.bbox,
+                        datetime_interval=self.datetime,
+                    )
+                extent_dict = collection.extent.to_dict()
+                collection_metadata = CollectionMetadata(
+                    catalog_url=self.base_url,
+                    id=collection.id,
+                    title=collection.title or "no title",
+                    spatial_extent=extent_dict["spatial"]["bbox"],
+                    temporal_extent=extent_dict["temporal"]["interval"],
+                    description=collection.description,
+                    keywords=collection.keywords or [],
+                    hint=hint,
+                )
+                results.append(collection_metadata)
+
+        return results
```

Comment:
```suggestion
    def get_collection_metadata(self) -> Iterable[CollectionMetadata]:
        catalog = Client.open(self.base_url)

        # add /collections conformance class just in case it's missing...
        # https://github.com/stac-utils/pystac-client/issues/320
        # cmr-stac is still missing this conformance class
        # https://github.com/nasa/cmr-stac/issues/236
        # this makes it possible to iterate through all collections
        catalog.add_conforms_to("COLLECTIONS")

        return (
            self.collection_metadata(collection)
            for collection in catalog.get_collections()
            if self.overlaps(collection)
        )

    def overlaps(self, collection: Collection) -> bool:
        return (
            self.spatially_overlaps(collection)
            and self.temporally_overlaps(collection)
            and self.textually_overlaps(collection)
        )

    def spatially_overlaps(self, collection: Collection) -> bool:
        return self.bbox is None or check_bbox_overlap(
            self.bbox,
            get_full_bounding_box(collection.extent.spatial.bboxes),  # type: ignore
        )

    def temporally_overlaps(self, collection: Collection) -> bool:
        start, end = collection.extent.temporal.intervals[0]

        return self.datetime is None or check_datetime_overlap(
            self.datetime, (ensure_utc(start), ensure_utc(end))
        )

    def textually_overlaps(self, collection: Collection) -> bool:
        text_fields: set[str] = {
            text
            for text in [
                collection.id,
                collection.title,
                collection.description,
                *(collection.keywords or []),
            ]
            if text
        }

        return not self.text or check_text_overlap(self.text, text_fields)

    def collection_metadata(self, collection: Collection) -> CollectionMetadata:
        hint = (
            generate_pystac_client_hint(
                base_url=self.base_url,
                collection_id=collection.id,
                bbox=self.bbox,
                datetime_interval=self.datetime,
            )
            if self.hint_lang == PYTHON
            else None
        )

        extent_dict = collection.extent.to_dict()

        return CollectionMetadata(
            catalog_url=self.base_url,
            id=collection.id,
            title=collection.title or "no title",
            spatial_extent=extent_dict["spatial"]["bbox"],
            temporal_extent=extent_dict["temporal"]["interval"],
            description=collection.description,
            keywords=collection.keywords or [],
            hint=hint,
        )
```